cangen.c

```diff
@@ -398,6 +407,32 @@ enum {
 	OPT_START = UCHAR_MAX + 2,
 };
 
+/*
+ * Search for CHAR_RANDOM in dataoptarg, save its position, replace it with 0.
+ * Return 1 if at least one CHAR_RANDOM found.
+ */
+static int parse_dataoptarg(char *dataoptarg, unsigned char *rand_position)
+{
+	int mode_format_selected = MODE_FIX;
+	int arglen = strlen(dataoptarg);
+
+	/* Mark nibbles with * as fuzzable */
+	for (int i=0; i < CANFD_MAX_DLEN && i < arglen/2; i++) {
+		if (optarg[2*i] == CHAR_RANDOM) {
+			optarg[2*i] = '0';
+			rand_position[i] += NIBBLE_H;
+			mode_format_selected = MODE_RANDOM_FIX;
+		}
+		if (optarg[2*i + 1] == CHAR_RANDOM) {
+			optarg[2*i + 1] = '0';
```

Comment:
dito

Reply:
Yes you are right. Missed that entirely.